### ISSUE TITLE
ExportModal allows filename with blanks to be exported and also has no maxLength restriction

### ISSUE THREAD

**Issue by naitikwrk** [carbon-design-system/ibm-products#4192]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [] Legacy/CDAI
- [] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.

Ideally, it would be nice if we can pass props to TextInput so that we can control properties like invalid, invalidText, maxLength etc. so that ExportModal behaviour can be customised as per consuming app.

**Comment by naitikwrk** [comment]
@elycheea yes. I have the fix on my local fork. I'm waiting for approval from my team before i can contribute here.

**Comment by davidmenendez** [comment]
@naitikwrk thanks for opening up this issue and opening a PR. i noticed that this is for v1. would you be interested in also opening a PR in the v2 (main) branch? i assume we want to include this bug fix in v2 @elycheea

**Comment by naitikwrk** [comment]
@davidmenendez sure
